wallet/src/de/schildbach/wallet/ui/dashpay/work/UpdateProfileWorker.kt

```diff
@@ -72,20 +72,27 @@ class UpdateProfileWorker(context: Context, parameters: WorkerParameters)
             }
         }
 
+        val avatarHash: ByteArray?
+
         // Perform the image upload here
         val avatarUrlToUpload = inputData.getString(KEY_LOCAL_AVATAR_URL_TO_UPLOAD)?:""
         val uploadService = inputData.getString(KEY_UPLOAD_SERVICE)?:""
         if (avatarUrlToUpload.isNotEmpty()) {
+            val avatarFile = File(avatarUrlToUpload)
+            @Suppress("BlockingMethodInNonBlockingContext")
+            avatarHash = Sha256Hash.of(avatarFile).bytes
             when (uploadService) {
                 EditProfileViewModel.ProfilePictureStorageService.GOOGLE_DRIVE.name -> {
-                    val avatarFileBytes = File(avatarUrlToUpload).readBytes()
+                    val avatarFileBytes = avatarFile.readBytes()
                     val fileId = saveToGoogleDrive(applicationContext, avatarFileBytes)
                     avatarUrl = "https://drive.google.com/uc?export=view&id=$fileId"
                 }
                 EditProfileViewModel.ProfilePictureStorageService.IMGUR.name -> {
                     //TODO:
                 }
             }
+        } else {
+            avatarHash = inputData.getByteArray(KEY_AVATAR_HASH)
         }
```

Comment:
this is leftover code from a previous story.  It is not currently used.

Reply:
What do you mean by that @HashEngineering?
Are you referring to a particular line or rather to the whole code?

Reply:
The entire block of code.  We are not currently performing the image upload with the `UpdateProfileWorker`.  As can be seen in this block, the `IMGUR` section has no code.

This line can be put outside this block:
`            avatarHash = inputData.getByteArray(KEY_AVATAR_HASH)`

Reply:
Fixed